private/buf/cmd/buf/command/push/push.go

```diff
@@ -347,6 +388,306 @@ func validateLabelFlagValues(flags *flags) error {
 	return nil
 }
 
+// We do not allow users to set --source-control-url, --create-default-label, and --label
+// flags if the --git-metadata flag is set.
+func validateGitMetadataFlags(flags *flags) error {
+	if flags.GitMetadata {
+		var usedFlags []string
+		if flags.SourceControlURL != "" {
+			usedFlags = append(usedFlags, sourceControlURLFlagName)
+		}
+		if len(flags.Labels) > 0 {
+			usedFlags = append(usedFlags, labelFlagName)
+		}
+		if flags.CreateDefaultLabel != "" {
+			usedFlags = append(usedFlags, createDefaultLabelFlagName)
+		}
+		if len(flags.Tags) > 0 {
+			usedFlags = append(usedFlags, tagFlagName)
+		}
+		if flags.Branch != "" {
+			usedFlags = append(usedFlags, branchFlagName)
+		}
+		if flags.Draft != "" {
+			usedFlags = append(usedFlags, draftFlagName)
+		}
+		if len(usedFlags) > 0 {
+			usedFlagsErrStr := strings.Join(
+				slicesext.Map(
+					usedFlags,
+					func(flag string) string { return fmt.Sprintf("--%s", flag) },
+				),
+				", ",
+			)
+			return appcmd.NewInvalidArgumentErrorf(
+				"The following flag(s) cannot be used in combination with --%s: %s.",
+				gitMetadataFlagName,
+				usedFlagsErrStr,
+			)
+		}
+	}
+	return nil
+}
+
+func getGitMetadataUploadOptions(
+	ctx context.Context,
+	container appext.Container,
+	flags *flags,
+) ([]bufmodule.UploadOption, error) {
+	input, err := bufcli.GetInputValue(container, flags.InputHashtag, ".")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Attempting to figure out a graceful way of surfacing an error if this is not a checkout of a Git repository (e.g. a source ref).

Reply:
We should validate for unchecked references here, so we can also use this to handle a `128`.